### PR TITLE
add gmLite and searchAgreement tests

### PR DESCRIPTION
**Pull request by Toyinccs**

### CODE REVIEW

File: src/main/java/com/scale/businessPages/AgreementPage.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    private WebElement openAllAccordion;
+
+
+    //
+//    @FindBy(xpath = "//*[@id='upcoming']")
+//    private WebElement upcomingFilter;
+
+
+    private String accordion = "button#accordion-default-heading-2[aria-expanded='true']";
+    private String allAccordionSections = "button.govuk-accordion__open-all[aria-expanded='true']";
```
Comment (amrutha-sireesh) [inline src/main/java/com/scale/businessPages/AgreementPage.java:48]: Can you move class variable declaration above functions?

File: src/main/java/com/scale/framework/utility/API/APIBase.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        return size;
+    }
+
+    public ArrayList<String> getAgreementName() {
+            ArrayList<String> countries = response.getBody().path("results.title");
+            for (int i = 0; i < countries.size(); i++) {
+                countries.get(i);
+            }
+            return countries;
+    }
```
Comment (amrutha-sireesh) [inline src/main/java/com/scale/framework/utility/API/APIBase.java:49]: Can you please format code?

File: src/main/java/com/scale/framework/utility/API/APIBase.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        int size = response.getBody().jsonPath().get("meta.total_results");
+        log.info(size);
+        return size;
+    }
+
+    public ArrayList<String> getAgreementName() {
+            ArrayList<String> countries = response.getBody().path("results.title");
+            for (int i = 0; i < countries.size(); i++) {
+                countries.get(i);
+            }
```
Comment (amrutha-sireesh) [inline src/main/java/com/scale/framework/utility/API/APIBase.java:47]: no need for this for loop here, right?

File: src/main/java/com/scale/framework/utility/API/APIBase.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        return response;
+    }
+
+    public Integer getStatusCode(){
+       int statusCode = response.statusCode();
+       return statusCode;
+    }
+    public int getTotalResults() {
+        int size = response.getBody().jsonPath().get("meta.total_results");
+        log.info(size);
```
Comment (amrutha-sireesh) [inline src/main/java/com/scale/framework/utility/API/APIBase.java:39]: Can you please add relevant info if you are using log statements

File: src/main/java/com/scale/businessPages/AgreementPage.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+
+public class AgreementPage extends Actions {
+    private WebDriver driver;
+    private ConfigurationReader configReaderObj;
+    private Logger log = Log.getLogger(ContactCCSPage.class);
```
Comment (amrutha-sireesh) [inline src/main/java/com/scale/businessPages/AgreementPage.java:23]: class name here is AgreementPage. I can see some class logger holds wrong class name across the repo :(

File: src/main/java/com/scale/businessPages/HomePage.java [hunk 6]
```diff
@@ -103,4 +117,31 @@ public void clickCheckBoxButton(String checkboxButtonName) {
...
+        WebElement element = driver.findElement(By.xpath(XPATH));
+        JavascriptExecutor executor = ((JavascriptExecutor) driver);
+        executor.executeScript("arguments[0].click();", element);
+        // element.click();
+        log.info("Buyer clicked on " + buttonName + " checkbox  button");
+        scenario.write(" Buyer clicked on " + buttonName + " checkbox button");
+    }
+
+    public void clickAgreementStatusFilter(String elementName){
+        String XPATH = "//*[@id= '"+ elementName +"']";
```
Comment (amrutha-sireesh) [inline src/main/java/com/scale/businessPages/HomePage.java:141]: can we follow variable name convention camel Case?

File: src/test/java/com/scale/stepdefs/SearchFrameworksPageSteps.java [hunk 7]
```diff
@@ -53,4 +63,64 @@ public void the_back_to_search_result_screen_is_opened_in_the_same_tab() {
         testContextObj.takeSnapShot();
     }
 
+    @When("I should see the search agreements results listed")
+    public void i_should_see_the_search_agreements_results_listed() {
+        apibase.getRequest("search-api/frameworks");
```
Comment (amrutha-sireesh) [inline src/test/java/com/scale/stepdefs/SearchFrameworksPageSteps.java:68]: can you please read the endpoint from from Config?

**Review by amrutha-sireesh — CHANGES_REQUESTED** [review]
Toyin, I have added few comments, no major changes identified. Thanks

File: src/main/java/com/scale/businessPages/AgreementPage.java (hunk 5 above)
Comment (ToyinOtak) [reply src/main/java/com/scale/businessPages/AgreementPage.java]: great spot, this has now been updated. Looks like the existing code has the wrong name conventional which I have also updated

**Pull request merged**